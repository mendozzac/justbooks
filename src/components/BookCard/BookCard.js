import "./BookCard.scss";
import Button from "../Button/Button";
//import { useContext } from "react";
//import BooksContext from "../../store/contexts/BooksContext";

const BookCard = ({ book, actionOnClick, addToFav }) => {

  return (
    <>
      <div className="card">
        <div onClick={actionOnClick}>
          <img
            className="card__image"
            src={book.volumeInfo.imageLinks.thumbnail}
            alt={book.volumeInfo.title}
            width="140"
            height="210"
          ></img>
          <div className="card__content">
            <h3 className="card__title">{book.volumeInfo.title}</h3>
            <p className="card__author">
              {" "}
              {book.volumeInfo.authors.slice(0, 2).join(" , ")}{" "}
            </p>{" "}
            <p className="card__price">
              {" "}
              {book.saleInfo.listPrice.amount + "â‚¬"}{" "}
            </p>
          </div>
        </div>
        <Button text="Add to favs" className="btn-warning card__fav-button" actionOnClick={() => addToFav(book)}
        />
      </div>
    </>
  );
};

export default BookCard;
