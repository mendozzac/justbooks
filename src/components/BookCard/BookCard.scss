.card {
  width: 140px;
  height: 284px;
  margin-bottom: 50px;
  border-style: none;
  &__info {
    padding-bottom: 0;
    font-size: 0.8em;
    &--title {
      font-weight: 800;
    }
  }
}
