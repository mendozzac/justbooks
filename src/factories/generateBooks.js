const faker = require("faker");

const generateBooks = () => {
  let books = [];

  for (let i = 1; i <= 10; i++) {
    let id = faker.random.alphaNumeric();
    let image = faker.image.image();
    let title = faker.name.title();
    let author = faker.name.firstName() + faker.name.lastName();
    let price = faker.commerce.price();

    books.push({
      id: id,
      image: image,
      title: title,
      author: author,
      price: price,
    });
  }

  return { items: books };
};

export default generateBooks;
