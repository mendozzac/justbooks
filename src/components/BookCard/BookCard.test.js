import { render, screen } from "@testing-library/react";
import BookCard from "./BookCard.js";
import generateBooks from "../../factories/generateBooks";
import { MemoryRouter } from "react-router-dom";
import ReactTestRenderer from "react-test-renderer";

describe("Given a BookCard component", () => {
  function mapFrom(bookData) {
    return {
      volumeInfo: {
        title: bookData.title,
        authors: [bookData.author],
        publisher: bookData.publisher,
        description: bookData.description,
        pageCount: bookData.pages,
        imageLinks: {
          thumbnail: bookData.image
            ? bookData.image
            : "https://images-na.ssl-images-amazon.com/images/I/51RTdGBiL6L._SX331_BO1,204,203,200_.jpg",
        },
      },
      saleInfo: {
        listPrice: {
          amount: "0.00",
        },
      },
    };
  }

  describe("When it receives a book", () => {
    test("Then it should render a card", () => {
      const books = generateBooks();
      const newBook = mapFrom(books.items[0]);

      render(
        <MemoryRouter initialEntries={["/home"]}>
          <BookCard book={newBook} />
        </MemoryRouter>
      );
      const bookCard = screen.getByRole("listitem");

      expect(bookCard).toBeInTheDocument();
    });
  });

  describe("When it recives and object", () => {
    test("then it should render a card with the object info inside", () => {
      const newBook = {
        volumeInfo: {
          title: "Harry",
          authors: ["Potter"],
          publisher: "Magia",
          description: "It's okay",
          pageCount: "100000",
          imageLinks: {
            thumbnail:
              "https://pbs.twimg.com/profile_images/3604358312/e784593ed966bc42476eab53261b8b09_400x400.jpeg",
          },
        },
        saleInfo: {
          listPrice: {
            amount: "0.00",
          },
        },
      };

      const cardComponent = ReactTestRenderer.create(
        <MemoryRouter initialEntries={["/home"]}>
          <BookCard book={newBook} />
        </MemoryRouter>
      );
      expect(cardComponent.toJSON()).toMatchSnapshot();
    });
  });
});
