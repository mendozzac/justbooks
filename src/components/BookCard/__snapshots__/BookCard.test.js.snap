// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a BookCard component When it recives and object then it should render a card with the object info inside 1`] = `
<li
  className="card"
>
  <div>
    <img
      alt="Harry"
      className="card__image"
      height="210"
      src="https://pbs.twimg.com/profile_images/3604358312/e784593ed966bc42476eab53261b8b09_400x400.jpeg"
      width="150"
    />
    <div
      className="card__content"
    >
      <h3
        className="card__title"
      >
        Harry
      </h3>
      <p
        className="card__author"
      >
         
        Potter
         
      </p>
       
      <p
        className="card__price"
      >
         
        0.00â‚¬
         
      </p>
    </div>
  </div>
  <button
    className="btn button btn-warning card__fav-button"
    disabled={false}
    onClick={[Function]}
  >
    Add to Favs
  </button>
</li>
`;
